include/llama.h

```diff
@@ -698,74 +698,82 @@ extern "C" {
                "Use llama_kv_self_seq_pos_max() and llama_kv_self_seq_pos_min() instead (https://github.com/ggml-org/llama.cpp/issues/13793)");
 
     // Clear the KV cache - both cell info is erased and KV data is zeroed
-    LLAMA_API void llama_kv_self_clear(
-                struct llama_context * ctx);
+    DEPRECATED(LLAMA_API void llama_kv_self_clear(
+                struct llama_context * ctx),
+            "Use llama_memory_clear() instead");
```

Comment:
Since we are adding a new function to do this, it would be good to add a parameter to make the buffer clear optional, since it adds a very large overhead. I don't remember why it was necessary.

Reply:
It's usually necessary because the K and V tensors are padded and can span over some old stale data that can cause some floating-point differences in the attention.